hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -5354,6 +5356,32 @@ private void replayFlushInStores(FlushDescriptor flush, PrepareFlushResult prepa
     }
   }
 
+  private long loadRecoveredHFilesIfAny(Collection<HStore> stores) throws IOException {
+    Path regionDir = getWALRegionDir();
+    long maxSeqId = -1;
+    for (HStore store : stores) {
+      String familyName = store.getColumnFamilyName();
+      FileStatus[] files =
+          WALSplitUtil.getRecoveredHFiles(fs.getFileSystem(), regionDir, familyName);
+      if (files != null && files.length != 0) {
+        for (FileStatus file : files) {
+          store.assertBulkLoadHFileOk(file.getPath());
```

Comment:
If the file is a corrupt HFile, an exception will be thrown here, which will cause the region to fail to open. 
Maybe we can add a new parameter to control whether to skip the exception, similar to recover edits which has a parameter "hbase.hregion.edits.replay.skip.errors";

Reply:
The skip errors only used for replay edits. It may lose data if there are a corrupt HFile?

Reply:
Let me open a follow-up issue for this.